scrapy/utils/datatypes.py

```diff
@@ -307,19 +308,48 @@ class LocalCache(collections.OrderedDict):
     """Dictionary with a finite number of keys.
 
     Older items expires first.
-
     """
 
     def __init__(self, limit=None):
         super(LocalCache, self).__init__()
         self.limit = limit
 
     def __setitem__(self, key, value):
-        while len(self) >= self.limit:
-            self.popitem(last=False)
+        if self.limit:
```

Comment:
This is a bugfix: without the `if` statement, it can fail with `TypeError: '>=' not supported between instances of 'int' and 'NoneType'`

Reply:
See #4123